docs/root/intro/arch_overview/upstream/aggregate_cluster.rst

```diff
@@ -0,0 +1,143 @@
+| 20%                   | 0%                    | 0%                    | 20%                   | 0%                    | 50%                | 50%                  |
++-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+--------------------+----------------------+
+| 0%                    | 0%                    | 0%                    | 100%                  | 0%                    | 0%                 | 100%                 |
++-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+--------------------+----------------------+
+| 0%                    | 0%                    | 0%                    | 72%                   | 0%                    | 0%                 | 100%                 |
++-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+--------------------+----------------------+
+
+Note: The above load balancing uses default :ref:`overprovisioning factor <arch_overview_load_balancing_overprovisioning_factor>` 
+which is 1.4 which means if 80% of the endpoints in a priority level are healthy, that level is 
+still considered fully healthy because 80 * 1.4 > 100.
+
+The example shows how the aggregate cluster level load balancer selects the cluster. E.g., healths 
+of {{20, 20, 10}, {25, 25}} would result in a priority load of {{28%, 28%, 14%}, {30%, 0%}} of 
+traffic. When normalized total health drops below 100, traffic is distributed after normalizing the 
+levels' health scores to that sub-100 total. E.g. healths of {{20, 0, 0}, {20, 0}} (yielding a 
+normalized total health of 56) would be normalized and each cluster will receive 20 * 1.4 / 56 = 50%
+of the traffic which results in a priority load of {{50%, 0%, 0%}, {50%, 0%, 0%}} of traffic.
+
+The load balancer reuses priority level logic to help with the cluster selection. The priority level
+logic works with integer health scores. The health score of a level is (percent of healthy hosts in 
+the level) * (overprovisioning factor), capped at 100%. P=0 endpoints receive level 0's health 
+score percent of the traffic, with the rest flowing to P=1 (assuming P=1 is 100% healthy - more on 
+that later). The integer percents of traffic that each cluster receives are collectively called the 
+system's "cluster priority load". For instance, for primary cluster, when 20% of P=0 endpoints are 
+healthy, 20% of P=1 endpoints are healthy, and 10% of P=2 endpoints are healthy; for secondary, when
+25% of P=0 endpoints are healthy and 25% of P=1 endpoints are healthy. The primary cluster will 
+receive 20% * 1.4 + 20% * 1.4 + 10% * 1.4 = 70% of the traffic. The secondary cluster will receive 
+min(100 - 70, 25% * 1.4 + 25% * 1.4) = 30% of the traffic.
```

Comment:
I assume we do min here because this this is done after normalizing, right? What if there are more than two clusters? What would the calculation be in the case? Would it be the the amount calculated during the normalizing process for the cluster in question?

Reply:
We do min because we don't want the sum of traffic exceed 100. If there are more than two cluster, the calculation would be similar. 

```
Primary: 20%, 20%, 10%; Secondary: 25%, 25%; Tertiary: 100%
Primary: 70%            Secondary: 30%       Tertiary: 0%
                                             ^ min(100 - 70 - 30, 100 * 1.4)

Primary: 20%, 20%, 10%; Secondary: 10%, 0%;                 Tertiary: 100%
Primary: 70%            Secondary: 14%                      Tertiary: 16%
^ min(100-0, 20*1.4+20*1.4+10*1.4)
                        ^ min(100-70, 10*1.4+0*1.4)         ^ min(100-70-14, 100*1.4)

Reply:
My main point is that you have to make sure that ultimately the traffic winds up at 100% total across all clusters, so this require some pre-compution of the normalized health, right? Because otherwise lets say that the last cluster doesn't have much healthy. In this case you might actually not have everything add up to 100?

Can you possibly just add a bit more text to make this clear for dense people like me? :)

Reply:
> My main point is that you have to make sure that ultimately the traffic winds up at 100% total across all clusters, so this require some pre-compution of the normalized health, right?

Correct! The normalized health score and priority load are pre computed. I will add some words here to make it more clear.